packages/sync/src/modules/Term_Relationships.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/**
+ * Term relationships sync module.
+ *
+ * @package automattic/jetpack-sync
+ */
+
+namespace Automattic\Jetpack\Sync\Modules;
+
+use Automattic\Jetpack\Sync\Defaults;
+use Automattic\Jetpack\Sync\Listener;
+use Automattic\Jetpack\Sync\Settings;
+
+/**
+ * Class to handle sync for term relationships.
+ */
+class Term_Relationships extends Module {
+	/**
+	 * Sync module name.
+	 *
+	 * @access public
+	 *
+	 * @return string
+	 */
+	public function name() {
+		return 'term_relationships';
+	}
+
+	/**
+	 * Initialize term relationships action listeners for full sync.
+	 *
+	 * @access public
+	 *
+	 * @param callable $callable Action handler callable.
+	 */
+	public function init_full_sync_listeners( $callable ) {
+		add_action( 'jetpack_full_sync_term_relationships', $callable, 10, 2 );
+	}
+
+	/**
+	 * Initialize the module in the sender.
+	 *
+	 * @access public
+	 */
+	public function init_before_send() {
+		// Full sync.
+		add_filter( 'jetpack_sync_before_send_jetpack_full_sync_term_relationships', array( $this, 'expand_term_relationships' ) );
+	}
+
+	/**
+	 * Enqueue the term relationships actions for full sync.
+	 *
+	 * @access public
+	 *
+	 * @todo This method has similarities with Automattic\Jetpack\Sync\Modules\Module::enqueue_all_ids_as_action. Refactor to keep DRY.
+	 * @see Automattic\Jetpack\Sync\Modules\Module::enqueue_all_ids_as_action
+	 *
+	 * @param array   $config               Full sync configuration for this sync module.
+	 * @param int     $max_items_to_enqueue Maximum number of items to enqueue.
+	 * @param boolean $state                True if full sync has finished enqueueing this module, false otherwise.
+	 * @return array Number of actions enqueued, and next module state.
+	 */
+	public function enqueue_full_sync_actions( $config, $max_items_to_enqueue, $state ) {
+		global $wpdb;
+		$items_per_page        = 1000;
+		$chunk_count           = 0;
+		$previous_interval_end = $state ? $state : array(
+			'object_id'        => '~0',
+			'term_taxonomy_id' => '~0',
+		);
+		$listener              = Listener::get_instance();
+		$action_name           = 'jetpack_full_sync_term_relationships';
+
+		// Count down from max_id to min_id so we get term relationships for the newest posts and terms first.
+		// phpcs:ignore WordPress.CodeAnalysis.AssignmentInCondition.FoundInWhileCondition, WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		while ( $ids = $wpdb->get_results( "SELECT object_id, term_taxonomy_id FROM $wpdb->term_relationships WHERE object_id <= {$previous_interval_end['object_id']} AND term_taxonomy_id < {$previous_interval_end['term_taxonomy_id']} ORDER BY object_id DESC, term_taxonomy_id DESC LIMIT {$items_per_page}", ARRAY_A ) ) {
+			// Request term relationships in groups of N for efficiency.
+			$chunked_ids = array_chunk( $ids, self::ARRAY_CHUNK_SIZE );
```

Comment:
The bit in the `while` is kind of the same as the one in `Automattic\Jetpack\Sync\Modules\Module::enqueue_all_ids_as_action`, so we can eventually refactor them to share the same method, but it feels like this can be done in a follow-up PR, hence the `@todo` in the docblock.